src/ipld.rs

```diff
@@ -16,14 +15,17 @@ use cid::Cid;
 #[derive(Clone, Debug)]
 #[non_exhaustive]
 pub enum AccessError {
```

Comment:
nit: I'd call this `IndexError`, or something like that. For me, "access error" indicates that I'm trying to "access" some service.

Reply:
That's a good one, thanks!